jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java

```diff
@@ -0,0 +1,242 @@
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static akka.pattern.PatternsCS.after;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class JmsConnectorsTest {
+
+    @Test
+    public void publishAndConsume() throws Exception {
+        withServer(ctx -> {
+            //#connection-factory
+            ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(ctx.url);
+            //#connection-factory
+
+            //#create-sink
+            Sink<String, NotUsed> jmsSink = JmsSink.create(
+                    JmsSinkSettings
+                            .create(connectionFactory)
+                            .withQueue("test")
+            );
+            //#create-sink
+
+            //#run-sink
+            List<String> in = Arrays.asList("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k");
+            Source.from(in).runWith(jmsSink, materializer);
+            //#run-sink
+
+            //#create-source
+            Source<String, NotUsed> jmsSource = JmsSource
+                    .textSource(JmsSourceSettings
+                            .create(connectionFactory)
+                            .withQueue("test")
+                            .withBufferSize(10)
+                    );
+            //#create-source
+
+            //#run-source
+            CompletionStage<List<String>> result = jmsSource
+                    .take(in.size())
+                    .runWith(Sink.seq(), materializer);
+            //#run-source
+
+            assertEquals(in, result.toCompletableFuture().get(3, TimeUnit.SECONDS));
+        });
+
+    }
+
+    @Test
+    public void applyingBackpressure() throws Exception {
+        withServer(ctx -> {
+            ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(ctx.url);
+            List<String> in = Arrays.asList("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k");
+            Source.from(in).runWith(JmsSink.create(JmsSinkSettings
+                    .create(connectionFactory)
+                    .withQueue("test")
+            ), materializer);
+
+            CompletionStage<List<String>> result = JmsSource
+                    .textSource(JmsSourceSettings
+                            .create(connectionFactory)
+                            .withQueue("test")
+                            .withBufferSize(1)
+                    )
+                    .mapAsync(1, e ->
+                            after(FiniteDuration.create(1, TimeUnit.SECONDS), system.scheduler(), system.dispatcher(), CompletableFuture.completedFuture(e))
+                    )
+                    .take(in.size()).runWith(Sink.seq(), materializer);
+
+            assertEquals(in, result.toCompletableFuture().get(15, TimeUnit.SECONDS));
+        });
+    }
+
+    @Test
+    public void deconnexionShouldFail() throws Exception {
```

Comment:
Also, the test cases that are not part of the docs are fine to just have in one version, java or scala, no need to run the exact same test from both languages.